rocAL/rocAL/source/meta_data/label_reader_folders.cpp

```diff
@@ -127,11 +127,13 @@ void LabelReaderFolders::read_all(const std::string& _path)
         filesys::path pathObj(subfolder_path);
         if(filesys::exists(pathObj) && filesys::is_regular_file(pathObj))
         {
-            // ignore files with extensions .tar, .zip, .7z
+            // ignore files with non-image extensions
             auto file_extension_idx = subfolder_path.find_last_of(".");
             if (file_extension_idx  != std::string::npos) {
                 std::string file_extension = subfolder_path.substr(file_extension_idx+1);
-                if ((file_extension == "tar") || (file_extension == "zip") || (file_extension == "7z") || (file_extension == "rar"))
+                std::transform(file_extension.begin(), file_extension.end(), file_extension.begin(),
+                    [](unsigned char c){ return std::tolower(c); });
+                if ((file_extension != "jpg") && (file_extension != "jpeg") && (file_extension != "png") && (file_extension != "ppm") && (file_extension != "bmp") && (file_extension != "pgm") && (file_extension != "tif") && (file_extension != "tiff") && (file_extension != "webp"))
```

Comment:
Can we decode files other than JPEGs like pgm and tiff etc? In that case shouldn't we ignore those?
I feel that we need to give another parameter thorugh file_reader which is a list of file filters(extensions). By default that can be set for jpegs. Can we add that paramter for file_reader as an optional paramter. If specified, those are the file extensions which will be accepted by the reader. Otherwise it will be ignored.

Reply:
The cv2 decoder can decode all of the file extensions specified in the file source reader and label_reader_folders.cpp hence we added all image extensions.
If we specify only jpeg files as the default extensions to be read using file reader, any dataset / samples with mixed image formats will be forced to only read jpeg files even if we can decode all the files using cv2 decoder.